src/env.cr

```diff
@@ -52,6 +47,31 @@ module ENV
     !!LibC.getenv(key)
   end
 
+  # Retrieve a value corresponding to a key. Raises a KeyError exception if the
+  # key does not exist.
+  #
+  def self.fetch(key)
+    fetch(key) do
+      raise KeyError.new "Missing ENV key: #{key.inspect}"
+    end
+  end
+
+  # Retrieve a value corresponding to a key. Return the second argument's value
+  # if the key does not exist.
+  #
+  def self.fetch(key, default)
+    fetch(key) { default }
+  end
+
+  # Retrieve a value corresponding to a key. Return the value of the block if
+  # the key does not exist.
+  #
+  def self.fetch(key)
+    value = self[key]?
+    return value if value
+    yield(key).to_s
```

Comment:
To add some safety, maybe type restrict the block: `def fetch(key : String, &block : String -> String?)`. You should still be able to use yield key afterwards.